src/libraries/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -915,22 +949,14 @@ public static Task<string[]> ReadAllLinesAsync(string path, Encoding encoding, C
 
         private static async Task<string[]> InternalReadAllLinesAsync(string path, Encoding encoding, CancellationToken cancellationToken)
         {
-            Debug.Assert(!string.IsNullOrEmpty(path));
-            Debug.Assert(encoding != null);
+            List<string> lines = new List<string>();
 
-            using (StreamReader sr = AsyncStreamReader(path, encoding))
+            await foreach (string line in InternalReadLinesAsync(path, encoding, cancellationToken))
```

Comment:
nit:
```suggestion
            await foreach (string line in InternalReadLinesAsync(path, encoding, cancellationToken).ConfigureAwait(false))
```
as before it was using with `ReadLineAsync`.